Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -644,6 +645,23 @@ public override void Reset(VCExpressionGenerator gen)
       }
     }
 
+    private void RecoverIfProverCrashedAfterReset()
+    {
+      resetCount += 1;
+      if (Process.GetExceptionIfProverDied() is Exception e)
+      {
+        if (resetCount == 1)
+        {
+          // First-time reset cannot be recovered
+          throw e;
+        }
+
+        Process.Close();
+        SetupProcess(true);
+        resetCount = 1;
```

Comment:
At the start of `RecoverIfProverCrashedAfterReset` this variable is incremented, so assigning 1 to it, would mean that next time it's already 2, right? If you want to reset this value shouldn't you set it to 0 ?

Reply:
Next time, it's already 2, right, but there is a good reason for that.
The code is catching errors that happen right after a `(reset)` command is issued, and no other errors. Currently, Boogie sends this command when the process starts for the first time, and that's the non-recoverable `(reset)` command that would be caught and thrown again.
Other than that, if a `(reset)` throw an exception after this case, it means that the prover already did some processing and it's really after that that the `(reset)` command failed. So we just want to restart the prover and not even send a `(reset)` command that would need to be checked again.

Sorry for the confusion, I could have just put a boolean there or just remove this counter alltogether (after all, if a reset at the beginning fails, there isn't anything that we could do)